host_core/lib/host_core/control_interface/acl.ex

```diff
@@ -1,23 +1,30 @@
 defmodule HostCore.ControlInterface.ACL do
   def all_actors() do
     HostCore.Actors.ActorSupervisor.all_actors()
-    |> Enum.map(fn {k, v} ->
-      %{
-        id: k,
-        revision: get_revision(k),
-        image_ref: get_image_ref(k),
-        instance_id: HostCore.Actors.ActorModule.instance_id(v)
-      }
+    |> Enum.flat_map(fn {id, pids} ->
+      revision = get_revision(id)
+      image_ref = get_image_ref(id)
+
+      pids
+      |> Enum.map(fn pid ->
+        %{
+          id: id,
+          revision: revision,
+          image_ref: image_ref,
+          instance_id: HostCore.Actors.ActorModule.instance_id(pid)
+        }
+      end)
     end)
   end
 
   def all_providers() do
+    # TODO: retrieve revision information for provider
     HostCore.Providers.ProviderSupervisor.all_providers()
     |> Enum.map(fn {pk, link, _contract, instance_id} ->
       %{
         id: pk,
         link_name: link,
-        revision: get_revision(pk),
+        revision: 0,
```

Comment:
I left this with a TODO as the claims information for providers is not being stored in the `:claims_table`. There's a couple different ways I could see solving for this, looking to get some thoughts:

1. store provider claims in the claims table under the provider_key as a key (different claims structure)
2. store the revision information as ProviderModule state
3. store the revision information in the `Registry.ProviderRegistry` along with instance id, pub key, etc.

I'm leaning towards one of these but curious if there are other thoughts.

Reply:
Option 1 is where I'm leaning (we should create a story to do this at some point). Option 3 is non-idiomatic as those kinds of registries are really designed to store pids.